compare.py

```diff
@@ -0,0 +1,204 @@
+        f_train, Pxx_train = signal.welch(train, sr, nperseg=1024)
+        f_pred, Pxx_pred = signal.welch(pred, sr, nperseg=1024)
+        
+        ax2.set_title('Power Spectral Density')
+        ax2.plot(f_dry, Pxx_dry, label='Dry', color='darkgray', linewidth=0.5, alpha=0.7)
+        ax2.plot(f_train, Pxx_train, label='Train', color='red', linewidth=0.5, alpha=0.6)
+        ax2.plot(f_pred, Pxx_pred, label='Pred', color='green', linewidth=0.5)
+        # Fix y axis based on train means identical graph scale across trainings
+        ax2.set_ylim([min(Pxx_train)*0.5, max(Pxx_train)*1.5])
+        ax2.set_xscale('log')
+        ax2.set_yscale('log')
+        ax2.set_xlim([20, 20000])
+        ax2.set_xticks([50, 100, 500, 1000, 5000, 10000, 20000])
+        ax2.set_xticklabels(['50', '100', '500', '1k', '5k', '10k', '20k'])
+        ax2.grid(which='both', axis='x', alpha=0.4)
+        ax2.grid(which='major', axis='y', alpha=0.4)
+        ax2.set_ylabel('Power Density')
+
+        # ==============================================================================
+        # Top right (AX3): dB Mel Spectrogram Diff
+        # ==============================================================================
+        ax3.set_title('dB Mel Spectrogram Diff')
+
+        f, t, sgram_pred = signal.spectrogram(pred, sr, nperseg=1024, noverlap=512, mode='magnitude')
+        f, t, sgram_train = signal.spectrogram(train, sr, nperseg=1024, noverlap=512, mode='magnitude')
+        sgram_pred_db = 10 * np.log10(sgram_pred + 1e-10)
+        sgram_train_db = 10 * np.log10(sgram_train + 1e-10)
+        sgram_diff_db = sgram_pred_db - sgram_train_db
+
+        # Plot the difference
+        ax3.pcolormesh(t, f, sgram_diff_db, shading='auto', cmap='rwg', vmin=-10, vmax=10)
+        ax3.set_ylabel('Frequency [Hz]')
+        ax3.set_xlabel('Time [sec]')
+        ax3.set_yscale('log')
+        ax3.set_ylim([20, 20000])
+        ax3.set_yticks([50, 100, 500, 1000, 5000, 10000, 20000])
+        ax3.set_yticklabels(['50', '100', '500', '1k', '5k', '10k', '20k'])
+        ax3.grid(which='both', axis='y', alpha=0.4)
+        # Add a legend for the color map
+        cbar = plt.colorbar(ax3.collections[0], ax=ax3, orientation='vertical')
+        cbar.set_label('Train+      Difference (dB)      Pred+')
+        cbar.ax.set_yticklabels([])
+
+        # ==============================================================================
+        # Bottom right (AX4): Waveform comparison
+        # ==============================================================================
+        overlap_waveform = np.minimum(pred, train)
+
+        ax4.set_title('Waveform Comparison')
+        ax4.plot(x_range, pred, label='Pred +', color='green', alpha=0.5)
+        ax4.plot(x_range, train, label='Train +', color='red', alpha=0.5)
+        ax4.plot(x_range, overlap_waveform, label='both', color='white', alpha=1)
+        ax4.plot(x_range, dry, label='Dry', color='lightgray', alpha=1)
```

Comment:
Hacky but kinda slick: maps both waveforms in diff colors but then maps their intersection in white to effectively erase everything that isn't the diff. This leaves room to map the dry signal in the same place.